USFMConverter/UI/Pages/FileView.axaml.cs

```diff
@@ -86,6 +89,30 @@ private async void OnBrowseClick(object? sender, RoutedEventArgs e)
             }
         }
 
+        private async void OnBrowseFilesClick(object? sender, RoutedEventArgs e)
+        {
+            var dialog = new OpenFileDialog();
+            dialog.AllowMultiple = true;
+            dialog.Title = "Select Files";
+
+            var extensions = CoreConverter.supportedExtensions
+                .Select(ex => ex.Replace(".", ""))
+                .ToList();
+
+            dialog.Filters.Add(new FileDialogFilter
+            {
+                Name = "USFM Documents",
+                Extensions = extensions
+            });
+
+            var paths = await dialog.ShowAsync((Window)this.VisualRoot);
+            var currentFileList = filesContainer.Items.Cast<string>();
```

Comment:
Perhaps you should change filesContainer.Items to a List, List implements IEnumerable so things wouldn't change. But that allows you to then do an AddRange() instead of this casting

Reply:
Will do in a subsequent PR